src/Build/BuildCheck/Infrastructure/CheckWrapper.cs

```diff
@@ -17,13 +18,34 @@ internal sealed class CheckWrapper
 {
     private readonly Stopwatch _stopwatch = new Stopwatch();
 
+    /// <summary>
+    /// Maximum amount of messages that could be sent per check rule.
+    /// </summary>
+    public const int MaxReportsNumberPerRule = 20;
+
+    /// <summary>
+    /// Keeps track of number of reports sent per rule.
+    /// </summary>
+    private int _reportsCount = 0;
+
+    /// <summary>
+    /// Flags that this check should no more used and be deregistered.
+    /// </summary>
+    public bool IsThrottled = false;
```

Comment:
```suggestion
    public bool IsThrottled { get; private set; } = false;
```

This should rather be autoproperty.
Fields should not be exposed.